src/lib/litegraph/src/types/widgets.ts

```diff
@@ -267,6 +269,7 @@ export interface IBaseWidget<
   /** Widget type (see {@link TWidgetType}) */
   type: TType
   value?: TValue
+  valueRef?: () => Ref<boolean | number | string | object | undefined>
```

Comment:
Might be able to get this to work if you use [NoInfer](https://www.totaltypescript.com/noinfer)